src/App.php

```diff
@@ -703,20 +700,20 @@ public function runFrontend(App\ModuleController $module, App\Router $router, IM
 			$page['page_title'] = $moduleName;
 
 			if (!$this->mode->isInstall() && !$this->mode->has(App\Mode::MAINTENANCEDISABLED)) {
-				$module = new ModuleController('maintenance', new Maintenance());
+				$module = new Maintenance();
 			} else {
 				// determine the module class and save it to the module instance
 				// @todo there's an implicit dependency due SESSION::start(), so it has to be called here (yet)
-				$module = $module->determineClass($this->args, $router, $this->config, $dice);
+				$module = $router->getModule();
```

Comment:
What about legacy modules still in `mod/`?

Reply:
It's now moved directly into the router. The router now didn't "just" find the right parameters & classes, it also creates it with the help of `Dice`, see the new `Router::getModule()` method